### ISSUE TITLE
zrep destroys resumable snapshots under some conditions

### ISSUE THREAD

**Issue by darkpixel** [bolthole/zrep#145]
```
root@useugsdnas01:~# zrep -t zrep-remote init tank/virt --redacted-- tank/backups/useugsd/virt
Setting zrep properties on tank/virt
Creating snapshot tank/virt@zrep-remote_000000
Sending initial replication stream to --redacted--:tank/backups/useugsd/virt
 188GiB 72:29:48 [1.30MiB/s] [                  <=>                           ]
--> Comcast outage here.  When it came back up I reattached to my screen session <--
packet_write_wait: Connection to 96.79.123.105 port 225: Broken pipe          ]
 608GiB 198:45:12 [ 892KiB/s] [              <=>                              ]
warning: cannot send 'tank/virt/vm-100-disk-0@zfs-auto-snap_02-2019-07-01-0700': signal received
ssh: Could not resolve hostname --redacted--: Temporary failure in name resolution
Destroying any zrep-related snapshots from tank/virt
Removing zrep-related properties from tank/virt
Error: Error transferring tank/virt@zrep-remote_000000 to --redacted--:tank/backups/useugsd/virt. Resetting
root@useugsdnas01:~# env | grep ZREP
ZREP_R=-R
ZREP_RESUME=yes
ZREP_SEND_FLAGS=--raw
ZREP_INC_FLAG=-i
ZREP_SKIP_EXPIRE=1
ZREP_OUTFILTER=pv -eIrabL 1M
root@useugsdnas01:~# 
```

As you can see, I have ```ZREP_RESUME=yes``` but zrep decided to trash the snapshot that already had 188 GB sent... ;)

Thankfully I was using the ```-R``` flag, so a bunch of intermediary snapshots were sent.  I'm going to sync the sub-datasets all to that point, then manually take a ```zrep-remote_000000``` snapshot for all of them and try ```sentsync```.

**Comment by ppbrown** [comment]
hmm
i thought i fixed that recently. are you using the latest latest?

**Comment by darkpixel** [comment]
Yeah.  Your last commit had the new 'HORRIBLY DANGEROUS' message in it.  (https://github.com/bolthole/zrep/commit/081f384fda73dc47a3b891d8e2d09c2814f4fbde#diff-dab9a2716958309b4ebe4876de935c2bR912)

I made sure my sources matched.

**Comment by ppbrown** [comment]
Doh.. the old reverse-logic bug.
fixed in 2506832 I think.

**Comment by darkpixel** [comment]
Thanks!  I won't be able to test this easily--I'll have to wait for connectivity issues.  But the code is so straight forward...  I'll get it pushed to all my boxes this evening.
